erlangen_nuernberg.py

```diff
@@ -0,0 +1,234 @@
+#!python3
+from urllib.request import urlopen
+from bs4 import BeautifulSoup as parse
+import re
+import datetime
+
+from pyopenmensa.feed import LazyBuilder
+
+from utils import Parser
+
+date_regex = re.compile('(?P<week_day>[A-Z][a-z])\ (?P<day>[0-9].)\.(?P<month>[0-9].)\.')
+price_regex = re.compile('(?P<val>[0-9]*,[0-9].)')
+
+roles = ('student', 'employee', 'other')
+
+def parse_url(url, today=False):
+    content = urlopen(url).read()
+    document = parse(content)
+    canteen = LazyBuilder()
+    table = document.find_all('table')[0]
+
+    def debug_print(food_type, food_description, priceing):
+            if(priceing is None):
+                print(date+': '+food_type+": "+food_description)
+            else:
+                print(date+': '+food_type+": "+food_description+" : ", end='')
+                for e in priceing:
+                    print(e, end=' ')
+                print()
+
+    def is_new_entry(tds):
+        td = tds[0]
+        return td.string is not None and date_regex.search(td.string) is not None
+
+    def is_end_of_entry(tds):
+        for td in tds:
+            if(td.string is None or td.string.strip() != ''):
+                return False
+        return True
+
+    def is_action_entry(td):
+        return td.text == 'Aktion'
+
+    def is_closed(tds):
+        return is_new_entry(tds) and get_pricing(tds, 4, 7) is None
+
+    def refactor_date(raw_date):
+        now = datetime.datetime.now()
+        day = date_regex.search(raw_date).group('day')
+        month = date_regex.search(raw_date).group('month')
+        year = now.year
+        if month == '01' and now.month == 12:
+            # if list depicts meals from this and the next year
+            year+=1
+        elif month == '12' and now.month == 1:
+            # if list depicts meals form this and the last year
+            year-=1
+        return day+'.'+month+'.'+str(year)
+
+    def parse_foot_type(td):
+        type = ''
+        if td.string is None:
+            img = td.find_all('img')[0]
+            src = img.get('src')
+            if('msc' in src):
+                type += 'Fish MSC '
+            elif('vegan' in src):
+                type += 'Vegan '
+        else:
+            if('R' in td.string):
+                type += 'Rind '
+            if('S' in td.string):
+                type += 'Schwein '
+            if('G' in td.string):
+                type += 'Geflügel '
+            if('V' in td.string):
+                type += 'Vegetarisch '
+            if('F' in td.string):
+                type += 'Fisch '
+            if('L' in td.string):
+                type += 'Lamm '
+            if('W' in td.string):
+                type += 'Wild '
+        return type
```

Comment:
Because you can have combinations off food. So There can bee a "Beef Pork" topic. I know "Beef Prok " is not that nice. If it causes trouble I'll remove that space at the end.